src/sgame/sg_bot_util.cpp

```diff
@@ -246,7 +246,7 @@ botEntityAndDistance_t BotGetHealTarget( const gentity_t *self )
 		// We skip buildings on roof or walls as bots can't wallwalk.
 		// Unless it's a booster, as boosters are often put on walls
 		// and have a large area of effect.
-		if ( candidate.ent && ( glm::dot( VEC2GLM(candidate.ent->s.origin2), up ) <= MIN_WALK_NORMAL || buildable == BA_A_BOOSTER ) )
+		if ( candidate.ent && ( glm::dot( VEC2GLM(candidate.ent->s.origin2), up ) >= MIN_WALK_NORMAL || buildable == BA_A_BOOSTER ) )
```

Comment:
```suggestion
		if ( candidate.ent && ( candidate.ent->s.origin2[2] >= MIN_WALK_NORMAL || buildable == BA_A_BOOSTER ) )
```

Reply:
And with that change you can remove the `static constexpr glm::vec3 up( 0.0f, 0.0f, 1.0f );` above.

Reply:
> That would be why the bots go to the OM very often

Changing this might actually be a problem. Assume no booster is present. If there are eggs or leeches far from base, most bots might go there to heal. Result: they will not defend their base. Until we implement better bot strategy, I would argue that bots should go to the overmind frequently. That is what I do myself.

> and sometime act weird on healing

That is the bug I am trying to fix. Currently, once the overmind is down, the bots stop to defend the rest of the base. Instead, they do strange jumps below the remaining eggs until fully healed.

I could simply add a comment to explain what the result of the dot product is. We might add a check for the vertical distance to the navmesh soon. Then we can refactor this. (Checking the vertical distance would also be useful for painsaw bots.)

Reply:
> > That would be why the bots go to the OM very often
> 
> Changing this might actually be a problem. Assume no booster is present. If there are eggs or leeches far from base, most bots might go there to heal. Result: they will not defend their base. Until we implement better bot strategy, I would argue that bots should go to the overmind frequently. That is what I do myself.

Actually overmind has higher priority compared to eggs and drills in all cases, and I don't think it can be built slanted, so it's probably always selected if available. This also means that in theory the bug you are afraid of shouldn't trigger. (Or overmind is dead).

And about the small change to make it more readable, it's ok if you want to refactor it another day.

(Btw, another thing I'm noticing now is that the code dosen't check if the buildable is reachable from navmesh. But that's not easy to fix, so should probably be left for another day)